### PR TITLE
Start of MIEngine refactor to support VS Code

### PR DESCRIPTION
**Pull request by gregg-miskelly**
This PR creates a new 'VSCodeRefactor' branch where we can workout changes needed to support VS Code in the MIEngine. This initial work is to create a new dll (Microsoft.DebugEngineHost) that will provide host services to debug engines. This hopefully will allow all the assemblies of the MIEngine to work in either VS or VS Code without needing to define yet-another-API-layer by swapping out just Microsoft.DebugEngineHost.

There is still a lot of work to go, but here is a start.

### CODE REVIEW

**Comment by gregg-miskelly** [comment]
FYI @caslan @orablu

File: src/MIDebugEngine/AD7.Impl/AD7ProgramProvider.cs [hunk 1]
```diff
@@ -28,6 +28,8 @@ int IDebugProgramProvider2.GetProviderProcessData(enum_PROVIDER_FLAGS Flags, IDe
         {
             processArray[0] = new PROVIDER_PROCESS_DATA();
 
+            Debug.Fail("Will this be called?");
```
Comment (chuckries) [inline src/MIDebugEngine/AD7.Impl/AD7ProgramProvider.cs]: Should this stay in here?

File: src/AndroidDebugLauncher/AndroidDebugLauncher.csproj [hunk 2]
```diff
@@ -137,6 +137,10 @@
     </Content>
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\DebugEngineHost\DebugEngineHost.csproj">
```
Comment (chuckries) [inline src/AndroidDebugLauncher/AndroidDebugLauncher.csproj]: Do the launchers use reference the wait loop? Is that why they need these project references?

**Comment by chuckries** [comment]
LGTM

File: src/AndroidDebugLauncher/AndroidDebugLauncher.csproj (hunk 2 above)
Comment (gregg-miskelly) [inline src/AndroidDebugLauncher/AndroidDebugLauncher.csproj]: Yes.

File: src/MIDebugEngine/AD7.Impl/AD7ProgramProvider.cs (hunk 1 above)
Comment (gregg-miskelly) [inline src/MIDebugEngine/AD7.Impl/AD7ProgramProvider.cs]: Good catch.

**Pull request merged**